### PR TITLE
[MRG]: tight bounds in contours

### LINKED ISSUE #1450: add_contours shrinks brain
Hi folks.
I have noticed a difference between `plot_anat` and `plot_roi` regarding the size in which the brain is displayed. With `plot_roi` it nearly fills the entire image, with `plot_anat` it is much smaller.
Is there a way to zoom in on the `plot_anat` images as well? Thanks.

`plot_roi`
plot_roi
`plot_anat`
plot_anat

### PR DESCRIPTION
**Pull request by GaelVaroquaux**
A test implementation for tight bounds in contours.

Attempts to fix #1450

### CODE REVIEW

**Comment by GaelVaroquaux** [comment]
The latest commit is a bit heuristic.

These changes will need to be war-tested on a bunch of different usage situations.

**Comment by KamalakerDadi** [comment]
I had a quick look at the plot_prob_atlas example. There is some problem with displaying contours.For instance, a msdl atlas in https://2495-1235740-gh.circle-artifacts.com/0/home/ubuntu/nilearn/doc/_build/html/auto_examples/01_plotting/plot_prob_atlas.html#sphx-glr-auto-examples-01-plotting-plot-prob-atlas-py

In plot_prob_atlas, we add an heuristic to get the threshold (levels) for displaying. May be we are double thresholding. I might also be wrong. I can have a closer look in more detail, later.

**Comment by GaelVaroquaux** [comment]
I changed the strategy to something that should be more robust.

I believe that this is now ready for merge.

@fliem, do you mind testing this again, as I changed the strategy. It should work fine.

File: doc/whats_new.rst [hunk 1]
```diff
@@ -33,6 +33,8 @@ Enhancements
 
     - Reworking and renaming of connectivity measures example. 
 
+    - Tighter bounding boxes when using add_contour for plotting.
```
Comment (KamalakerDadi) [inline doc/whats_new.rst]: add_contour -> add_contours. Just to be consistence with naming of our method.

File: nilearn/plotting/displays.py [hunk 2]
```diff
@@ -658,11 +658,22 @@ def _map_show(self, img, type='imshow',
...
+                thr = 1e-6
+            not_mask = np.logical_or(data > thr, data < -thr)
+            xmin_, xmax_, ymin_, ymax_, zmin_, zmax_ = \
+                get_mask_bounds(new_img_like(img, not_mask, affine))
+        elif hasattr(data, 'mask') and isinstance(data.mask, np.ndarray):
             not_mask = np.logical_not(data.mask)
             xmin_, xmax_, ymin_, ymax_, zmin_, zmax_ = \
                 get_mask_bounds(new_img_like(img, not_mask, affine))
 
+
```
Comment (KamalakerDadi) [inline nilearn/plotting/displays.py]: Just to avoid PEP8. Can you please remove this ?

**Comment by KamalakerDadi** [comment]
Do you have any tests to add ?

**Comment by GaelVaroquaux** [comment]
> Do you have any tests to add ?

I couldn't think of a way to add a test here.

**Comment by bthirion** [comment]
LGTM.

**Comment by fliem** [comment]
Output still looks good.

**Comment by GaelVaroquaux** [comment]
> Output still looks good.

Great. Once I get @KamalakerDadi's +1, we'll merge.

**Comment by KamalakerDadi** [comment]
+1 from my side. Merging this one.

**Pull request merged**